src/app/utils/field-validation/validators/numberValidator.ts

```diff
@@ -59,29 +62,64 @@ const maxLengthValidator: NumberValidatorConstructor =
 const exactLengthValidator: NumberValidatorConstructor =
   (numberField) => (response) => {
     const { answer } = response
-    const { customVal } = numberField.ValidationOptions
+    const { customVal } = numberField.ValidationOptions.LengthValidationOptions
     return !customVal || answer.length === customVal
       ? right(response)
       : left(`NumberValidator:\t answer does not match custom exact length`)
   }
 
 /**
- * Returns the appropriate validation function
- * based on the number validation option selected.
+ * Returns the appropriate number length validation function
+ * based on the number length validation option selected.
  */
 const getNumberLengthValidator: NumberValidatorConstructor = (numberField) => {
-  switch (numberField.ValidationOptions.selectedValidation) {
-    case NumberSelectedValidation.Min:
+  switch (
+    numberField.ValidationOptions.LengthValidationOptions
+      .selectedLengthValidation
+  ) {
+    case NumberSelectedLengthValidation.Min:
       return minLengthValidator(numberField)
-    case NumberSelectedValidation.Max:
+    case NumberSelectedLengthValidation.Max:
       return maxLengthValidator(numberField)
-    case NumberSelectedValidation.Exact:
+    case NumberSelectedLengthValidation.Exact:
       return exactLengthValidator(numberField)
     default:
       return right
   }
 }
 
+/**
+ * Returns a validation function to check if number is
+ * within the number range specified.
+ */
+const getNumberRangeValidator: NumberValidatorConstructor =
```

Comment:
Since this actually does the base logic of performing the validation, we should follow the same function naming convention as the other base validators a la `exactLengthValidator`. Can we change the name of this function to `rangeValidator`?